### DISCUSSION TITLE
Guide to disable typescript and remove dependencies

### DISCUSSION THREAD

**Discussion by kaikaiiiiiii** [cawa-93/vite-electron-builder#339] [Show and tell]
Any guide about it?

Humble request from a beginner to electron app, without any knowledge about typescript.

**Comment by cawa-93** [comment]
In general, all you need to do is rewrite the source files from TypeScript to JavaScript. There is no need to change the configuration of the bundler. I have reviewed this template and provide a list of necessary steps to completely remove all traces of TS. Most of these steps are optional and change the configuration of the IDE, linters, CI, and so on. I have highlighted important steps.

In root:
1. Remove `types` directory


In `.github`
1. Remove `.github/workflows/typechecking.yml`
2. Remove `typechecking`  job and mentions from `needs` and `paths-ignore` in `.github/workflows/ci.yml`
3. Remove all `ts` mentions in `.github/workflows/lint.yml`

In `package.json`
1. Remove all `typecheck:*` sctipts 
3. Remove all `ts` mentions from `lint` and `format` scripts
4. Remove unnecessary dependencies by:
    ```
    npm uninstall @typescript-eslint/eslint-plugin typescript vue-tsc
    ```

In `.nano-staged.mjs` remove typechicking rule
https://github.com/cawa-93/vite-electron-builder/blob/1f007c89d2f454e2aa3c13a2d6a8ab295a6a280a/.nano-staged.mjs#L6-L20

In `.eslintrc.json` Remove:
- `plugin:@typescript-eslint/recommended` from `extends`
- `parser`
- `parserOptions`
- `@typescript-eslint` from `plugins`
- All `@typescript-eslint/*` from `rules`

In `packages/main`:
1. **Rewrite `src/index.ts` to `src/index.js`**
1. **In `vite.config.js` change `build.lib.entry` to `src/index.js`**
3. **Rewrite tests**
1. Remove `tsconfig.json`

In `packages/preload`:
1. **Rewrite `src/index.ts` to `src/index.js`**
1. **In `vite.config.js` change `build.lib.entry` to `src/index.js`**
2. **Rewrite tests**
1. Remove `tsconfig.json`
2. Remove `exposedInMainWorld.d.ts ` if exist

In `packages/renderer`:
1. **Rewrite all files in `src` from TS to JS. If you want use Vue, you need remove `lang="ts"` attribute in all `.vue` files.**
2. **In `index.html` change script `src` attribute to you new entry file (it may be `./src/index.js` for example).**
2. **Rewrite tests**
4. Remove `tsconfig.json`
5. In `.eslintrc.json` remove `parserOptions.parser`
6. Remove `types` directory

Maybe I missed something, so you will have to perform the further configuration yourself.

I do not guarantee that this list will remain relevant in the future.

**Note about tests:**
In some tests may be used `vi.mocked(SomeObject)`. This just a TS-helper that actually do nothing and may be safely replaced by `SomeObject`

**Reply by JoCat** [reply]
> 4. Remove `tsconfig.json` directory

file :smiley:

**Reply by cawa-93** [reply]
@JoCat It seemed to you 😇

**Reply by cawa-93** [reply]
Upd: Added `dts-for-context-bridge` as ts-specific dependency.

> Remove unnecessary dependencies by: 
> ```
> npm uninstall @types/electron-devtools-installer @typescript-eslint/eslint-plugin typescript vue-tsc dts-for-context-bridge
> ```

**Reply by cawa-93** [reply]
Upd:
Add note about tests

**Reply by cawa-93** [reply]
Updated

**Reply by cawa-93** [reply]
Updated

**Comment by EvanBldy** [comment]
If you want I have made a fork without typescript : https://github.com/EvanBldy/vite-electron-builder

**Reply by cawa-93** [reply]
If you plan to maintain your fork you can share it [here](https://github.com/cawa-93/vite-electron-builder/discussions/categories/forks)

**Reply by aisiklar** [reply]
@EvanBldy @cawa-93  
Hi Guys,
First thanks for these projects and happy new year to you all.
I am thinking of starting a react-electron (with JS) project. 
Evan, can you kindly give info about the current status of your fork? I am asking bc it was last updated in 2021.
Cawa, is there a boilerplate for JS (and vite) lang? 
I don't know TS, though I plan to get into this later time, it is currently not the appropriate time now...
Thanks, kind regards

**Reply by cawa-93** [reply]
> Cawa, is there a newer boilerplate for js lang?

I Trying track all forks here: https://github.com/cawa-93/vite-electron-builder/discussions/categories/forks

**Reply by aisiklar** [reply]
thanks for the quick response.

**Reply by EvanBldy** [reply]
Hi @cawa-93 @aisiklar,
Sorry I have no time to maintain my fork anymore :/
